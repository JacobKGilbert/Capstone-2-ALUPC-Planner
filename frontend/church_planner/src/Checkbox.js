import { useState } from 'react'
import { Input } from 'reactstrap'

const Checkbox = ({ defCheck, idAndName, updateFormData }) => {
  const [ checked, setChecked ] = useState(defCheck)

  const handleChange = (evt) => {
    updateFormData(evt)
    setChecked(!checked)
  }

  return (
    <Input
      type="checkbox"
      id={`${idAndName}`}
      name={`${idAndName}`}
      value={checked}
      onChange={handleChange}
      checked={checked}
    />
  )
}

export default Checkbox