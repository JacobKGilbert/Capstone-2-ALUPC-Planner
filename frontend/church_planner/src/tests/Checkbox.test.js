import React from 'react'
import { render } from '@testing-library/react'
import Checkbox from '../Checkbox'

it('should render', () => {
  render(<Checkbox />)
})

it('should match snapshot', () => {
  const { asFragment } = render(<Checkbox />)
  expect(asFragment()).toMatchSnapshot()
})
