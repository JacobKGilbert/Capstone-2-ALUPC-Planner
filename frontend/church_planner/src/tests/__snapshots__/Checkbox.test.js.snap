// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <input
    class="form-check-input"
    id="undefined"
    name="undefined"
    type="checkbox"
    value=""
  />
</DocumentFragment>
`;
